bindings/c/src/types.rs

```diff
@@ -178,3 +180,63 @@ impl opendal_metadata {
         }
     }
 }
+
+/// [`opendal_operator_options`] represents a series of string type key-value pairs, it may be used for initialization
+#[repr(transparent)]
+pub struct opendal_operator_options {
+    pub inner: *mut HashMap<String, String>,
```

Comment:
We could use pub(crate)

Reply:
It looks good to me. I just want to know is this required by FFI?

Reply:
This is not required, I have changed it to pub(crate) now.

Reply:
> This is not required, I have changed it to pub(crate) now.

My point is can we make it priviate? And can we remove `HashMap_String__String` from our C API?

Reply:
> My point is can we make it private?

I access the keys and values in the `opendal_operator_new()`, so a pub(crate) will make it easier.

> And can we remove `HashMap_String__String` from our C API?

Then we have to implement our own hashmap or include third-party dependencies, which is not what we want I think.

If we really really hate the `HashMap_String__String`, we should use the original solution of `opendal_kv` (maybe another name).

Reply:
I'm bit confused about that. It's ok to use `HashMap` in `opendal_operator_options`. But we expose a public struct call `opendal_operator_options` in C with its own API. Why we need to expose `HashMap_String__String` too?

Reply:
> I access the keys and values in the `opendal_operator_new()`, so a pub(crate) will make it easier.

Oh, I got it. It's better to add a API `into_inner` (in Rust only) for `opendal_operator_options` instead access inner directly.

Reply:
> I'm bit confused about that. It's ok to use `HashMap` in `opendal_operator_options`. But we expose a public struct call `opendal_operator_options` in C with its own API. Why we need to expose `HashMap_String__String` too?

Inside the opendal_operator_options, it is a pointer to a HashMap<String, String>.
When we compile the opendal library, we actually compile it along with the Ruststd (since it is a Rust code).
Therefore, the compiled library will contains a type called `HashMap_String__String`.

Reply:
> > I access the keys and values in the `opendal_operator_new()`, so a pub(crate) will make it easier.
> 
> Oh, I got it. It's better to add a API `into_inner` (in Rust only) for `opendal_operator_options` instead access inner directly.

Maybe not `into_inner`, since it will consume the HashMap. We could just have a `as_ref`

Reply:
> When we compile the opendal library, we actually compile it along with the Ruststd (since it is a Rust code).

Thanks a lot! Lesson learnt.